tsl/test/expected/split_chunk.out

```diff
@@ -1163,3 +1163,76 @@ select count(*) from :compress_relid2;
      0
 (1 row)
 
+---
+--- Test splitting chunks that are using a time partitioning function
+---
+select p.oid as time_funcid
+from pg_proc p
+join pg_namespace n on (p.pronamespace = n.oid)
+where proname = 'to_timestamp'
+and provolatile = 'i'
+and n.nspname = 'pg_catalog' \gset
```

Comment:
Suggest to add a comment explaining why you do this lookup instead of using the function name directly. (I think the lookup inside `create_hypertable` is broken here and should only give an error if there is *no* viable candidate.)

Reply:
There are multiple versions of to_timestamp so need to lookup the exact one. Will add comment.

Reply:
> There are multiple versions of to_timestamp so need to lookup the exact one. Will add comment.

Yup, I had to test this and discovered the issue. I think that in reality the lookup should check for an immutable function with the right signature and only throw an error if it finds none.

Something for a rainy day.